### PR TITLE
Remove manual dynamic allocations in Xcast

### PR DESCRIPTION
**Pull request by payetvin**
#2244

### CODE REVIEW

File: src/libs/antares/study/area/list.cpp [hunk 1]
```diff
@@ -449,8 +449,7 @@ const AreaLink* AreaList::findLink(const AreaName& area, const AreaName& with) c
 
 void AreaList::clear()
 {
-    delete[] byIndex;
-    byIndex = nullptr;
+    byIndex.clear();
```
Comment (guilpier-code) [inline src/libs/antares/study/area/list.cpp:452]: I guess that the **delete** instruction in : 
```C++
for (auto i = copy.begin(); i != end; ++i)
{
    delete i->second;
}
```
would require some work, and cannot be done in this one, right ?

File: src/libs/antares/study/area/list.cpp [hunk 2]
```diff
@@ -449,8 +449,7 @@ const AreaLink* AreaList::findLink(const AreaName& area, const AreaName& with) c
...
 
     if (!areas.empty())
```
Comment (guilpier-code) [inline src/libs/antares/study/area/list.cpp]: Note that tests like : 
```C++
if (!areas.empty())
{
    ...
}
```
are useless. 
I guess we could remove them safely (keeping the inner block, of course), right ?

File: src/solver/ts-generator/xcast/xcast.cpp [hunk 3]
```diff
@@ -58,11 +58,6 @@ XCast::XCast(Data::Study& study, Data::TimeSeriesType ts, IResultWriter& writer)
 {
 }
 
-XCast::~XCast()
```
Comment (guilpier-code) [inline src/solver/ts-generator/xcast/xcast.cpp:61]: About class **XCast**'s constructor : 
```c++
    pNeverInitialized(true),
    pAccuracyOnCorrelation(false),
```
We should prefer initializing these data members inside class declaration

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/solver/ts-generator/xcast/xcast.cpp (hunk 3 above)
Comment (payetvin) [reply src/solver/ts-generator/xcast/xcast.cpp:61]: Agreed, will do

File: src/libs/antares/study/area/list.cpp (hunk 1 above)
Comment (payetvin) [reply src/libs/antares/study/area/list.cpp:452]: Indeed the Area object is used in more part of the code, I chose to stop this PR here before it become to much

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**